File: generate-icons.js
const fs = require('fs');
const { createCanvas } = require('canvas');

function createIcon(size) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // Create gradient background
    const gradient = ctx.createLinearGradient(0, 0, size, size);
    gradient.addColorStop(0, '#667eea');
    gradient.addColorStop(1, '#764ba2');
    
    // Background
    ctx.fillStyle = gradient;
    ctx.fillRect(0, 0, size, size);
    
    // Add text
    ctx.fillStyle = 'white';
    ctx.font = `bold ${Math.floor(size * 0.3)}px Arial`;
    ctx.textAlign = 'center';
    ctx.textBaseline = 'middle';
    ctx.fillText('RT', size/2, size/2);
    
    // Add accent for larger icons
    if (size >= 48) {
        ctx.fillStyle = 'rgba(255, 255, 255, 0.8)';
        ctx.beginPath();
        ctx.arc(size * 0.75, size * 0.25, size * 0.08, 0, 2 * Math.PI);
        ctx.fill();
    }
    
    return canvas;
}

// Generate icons
const sizes = [16, 48, 128];
sizes.forEach(size => {
    const canvas = createIcon(size);
    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync(`icons/icon${size}.png`, buffer);
    console.log(`Generated icon${size}.png`);
});

console.log('All icons generated successfully!');

File: generate_icons.py
from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size):
    # Create a new image with transparent background
    image = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(image)
    
    # Create gradient effect with solid colors
    # Background with gradient-like effect
    for i in range(size):
        for j in range(size):
            # Create gradient from purple to blue
            r = int(102 + (118 - 102) * i / size)  # 667eea to 764ba2
            g = int(126 + (75 - 126) * i / size)
            b = int(234 + (162 - 234) * i / size)
            draw.point((i, j), (r, g, b, 255))
    
    # Add text
    try:
        # Try to use a system font
        font_size = max(8, size // 4)
        font = ImageFont.truetype("/System/Library/Fonts/Arial.ttf", font_size)
    except:
        # Fallback to default font
        font_size = max(8, size // 4)
        font = ImageFont.load_default()
    
    # Calculate text position
    text = "RT"
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    x = (size - text_width) // 2
    y = (size - text_height) // 2
    
    # Draw text
    draw.text((x, y), text, fill=(255, 255, 255, 255), font=font)
    
    # Add accent dot for larger icons
    if size >= 48:
        accent_x = int(size * 0.75)
        accent_y = int(size * 0.25)
        accent_radius = max(2, int(size * 0.08))
        draw.ellipse([accent_x - accent_radius, accent_y - accent_radius, 
                     accent_x + accent_radius, accent_y + accent_radius], 
                    fill=(255, 255, 255, 200))
    
    return image

# Create icons directory if it doesn't exist
os.makedirs('icons', exist_ok=True)

# Generate icons
sizes = [16, 48, 128]
for size in sizes:
    icon = create_icon(size)
    icon.save(f'icons/icon{size}.png', 'PNG')
    print(f'Generated icon{size}.png')

print('All icons generated successfully!')
